### PR TITLE
Feature: added ability to compose with basic auth storybooks

### PR DESCRIPTION
**Pull request by enterframe**
Issue:
When using storybooks composition feature, it was not possible to compose storybooks secured by basic-auth.

## What I did
I just used the default style to predefine basic auth credentials in an url and proxied them forward to the fetch request

## How to test
In the sandbox template define a basic auth secured ref, it will successfully show up in the storybook refs pane

**/.storybook/main.cjs**
```
...
  "refs": {
    "shared-components": {
      "title": "Shared Components",
      "url": "https://user:pass@www.my-shared-components.com/"
    },
  },
...
```

- [x] Is this testable with Jest or Chromatic screenshots?
- [ ] Does this need a new example in the kitchen sink apps?
- [ ] Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

**Comment by shilman** [comment]
cc @ndelangen @tmeasday

File: code/lib/manager-api/src/modules/refs.ts
Comment (ndelangen) [inline code/lib/manager-api/src/modules/refs.ts:165]: Could you place this code in a function that returns:
```ts
interface UrlParseResult {
  url: string;
  authorization: string | undefined;
}
```

File: code/lib/manager-api/src/modules/refs.ts
Comment (ndelangen) [inline code/lib/manager-api/src/modules/refs.ts:182]:
```suggestion
      const headers: HeadersInit = {
        Accept: 'application/json',
      };

      if (urlParseResult.authorization) {
        Object.assign(headers, {
          Authorization: `Basic ${urlParseResult.authorization}`,
        });
      }
```

Comment (ndelangen) [reply code/lib/manager-api/src/modules/refs.ts:182]: I think this is great, thank you for the contribution @enterframe

**Comment by ndelangen** [comment]
@enterframe I'd like to update this branch with changes upstream but you've blocked maintainers from pushing to your fork, can you please update your branch yourself? Or allow me to do it, and ping me?

I'd like to merge this PR, but it's blocked on the above, 🙏 thanks!

**Comment by enterframe** [comment]
> @enterframe I'd like to update this branch with changes upstream but you've blocked maintainers from pushing to your fork, can you please update your branch yourself? Or allow me to do it, and ping me?
> 
> I'd like to merge this PR, but it's blocked on the above, 🙏 thanks!
> 
> Screenshot 2023-01-09 at 21 00 05 Which is weird because it looks like I was able to do it before? 😕

@ndelangen I did not change any settings in the fork and yes you updated it successfully before 🤷‍♂️ I do not see any changes upstream but for safety's sake I added you as a maintainer.. 🤘

**Review by ndelangen — APPROVED** [review]

**Pull request merged**
